src/cat.js

```diff
@@ -37,6 +43,34 @@ function _cat(options, files) {
     cat += fs.readFileSync(file, 'utf8');
   });
 
+  if (options.number) {
+    cat = addNumbers(cat);
+  }
+
   return cat;
 }
 module.exports = _cat;
+
+function addNumbers(cat) {
+  var addEOF = false;
+  var lines = cat.split('\n');
+
+  if (lines[lines.length - 1] === '') {
+    addEOF = true;
+    lines.splice(-1);
+  }
+
+  lines = lines.map(function (line, i) {
+    return number(i + 1) + ' ' + line;
```

Comment:
I think GNU cat uses a tab, not a single space, between the number and line. It also makes for slightly easier parsing. I'm not sure if this is specified by POSIX, however.